multiversx/mx-chain-go
# Save `synced` miniblocks at bootstrap

## Reasoning behind the pull request
- In some system tests we get this `log.WARN` 
`WARN [2023-01-25 10:30:08.939]   shardProcessor.indexBlockIfNeeded cannot prepare argSaveBlock error = transactionsFeeProcessor.PutFeeAndGasUsed key 8055c3312da097de8bf0c462f3be53316b81063e330316d939edb79a17384510 not found in Transactions. `. This `log.Warn` appears because at the bootstrap `miniblocks` that are applied are not saved in storage. 
## Proposed changes
-  In order to fix this problem `miniblocks` have to be saved in the storage. When a node will start with the flag `StartInEpochEnabled` it will save the miniblocks that are applied in storage.

## Testing procedure
- Run a system-test and ensure that the log from the description will no longer appear.

## Pre-requisites

Based on the [Contributing Guidelines](https://github.com/multiversx/mx-chain-go/blob/master/.github/CONTRIBUTING.md#branches-management) the PR author and the reviewers must check the following requirements are met:
- was the PR targeted to the correct branch?
- if this is a larger feature that probably needs more than one PR, is there a `feat` branch created?
- if this is a `feat` branch merging, do all satellite projects have a proper tag inside `go.mod`?
